server/lib/message-sender.js

```diff
@@ -504,13 +516,21 @@ class MessageSender {
         try {
             result = await this._sendMessage({listId: campaignMessage.list, subscriptionId: campaignMessage.subscription});
         } catch (err) {
-            await knex('campaign_messages')
+            if (err.code === CampaignMessageErrorCode.NORETRY) {
+              await knex('campaign_messages')
+                .where({id: campaignMessage.id})
+                .update({
+                  status: CampaignMessageStatus.FAILED,
```

Comment:
This will add status = -1 to mysql and I'm not really sure about the implications of that as of now. My hope is the current code will not try to send that message anymore. We should add some code that will allow us to present errors for failed messages in the UI as well as the ability to Retry Failed Messages.

Reply:
Indeed. It is necessary to go though the code and look for CampaignMessageStatus and figure out where the FAILED needs to be added.